src/main/java/carpet/script/language/ControlFlow.java

```diff
@@ -42,26 +44,80 @@ public static void apply(Expression expression) // public just to get the javado
 
         expression.addFunction("exit", (lv) -> { throw new ExitStatement(lv.size()==0?Value.NULL:lv.get(0)); });
 
-        expression.addFunction("throw", (lv)-> {throw new ThrowStatement(lv.size()==0?Value.NULL:lv.get(0)); });
+        expression.addLazyFunction("throw", -1, (c, t, lv)-> 
+        {
+            switch (lv.size()) 
+            {
+                case 0:
+                    c.host.issueDeprecation("throw() (without id)");
```

Comment:
throwing a 'null' by default is actually pretty useful, especially if you don't care what you pass. makes it like a more generalized 'break()'

Reply:
Will revert that then.